ts/client/src/client.ts

```diff
@@ -1248,33 +1248,41 @@ export class MangoClient {
     let wrappedSolAccount: Keypair | undefined;
     let preInstructions: TransactionInstruction[] = [];
     let postInstructions: TransactionInstruction[] = [];
-    const additionalSigners: Signer[] = [];
     if (mintPk.equals(NATIVE_MINT)) {
-      wrappedSolAccount = new Keypair();
+      // Generate a random seed for wrappedSolAccount.
+      const seed = Keypair.generate().publicKey.toBase58().slice(0, 32);
+      // Calculate a publicKey that will be controlled by the `mangoAccount.owner`.
+      const wrappedSolAccount = await PublicKey.createWithSeed(
```

Comment:
This is where the bug happens, I'm creating a new `const` rather than assigning to the `let` defined above, on line 1248 🤦